compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/ClassReference.kt

```diff
@@ -221,164 +212,7 @@ public sealed class ClassReference : Comparable<ClassReference>, AnnotatedRefere
-     * as if it's generated), then the function can only be parsed to have the projected types
-     * of `P1` and `R`
-     *
-     * @receiver The class which actually references the type. In the above example, this
-     * would be `SomeFactory`.
-     * @param declaringClass The class which declares the generic type name. In the above
-     * example, this would be `Function1`.
-     * @param typeToResolve The generic reference to be resolved. In the above example, this
-     * is the `T` **return type**.
-     */
-    public fun resolveGenericKotlinTypeOrNull(
-      declaringClass: ClassReference,
-      typeToResolve: KotlinType
-    ): KtTypeReference? {
-      val parameterKotlinType = when (typeToResolve) {
-        is FlexibleType -> {
-          // A FlexibleType comes from Java code where the compiler doesn't know whether it's a
-          // nullable or non-nullable type. toString() returns something like "(T..T?)". To get
-          // proper results we use the type that's not nullable, "T" in this example.
-          if (typeToResolve.lowerBound.isMarkedNullable) {
-            typeToResolve.upperBound
-          } else {
-            typeToResolve.lowerBound
-          }
-        }
-        is DefinitelyNotNullType -> {
-          // This is known to be not null in Java, such as something annotated `@NotNull` or
-          // controlled by a JSR305 or jSpecify annotation.
-          // This is a special type and this logic appears to match how kotlinc is handling it here
-          // https://github.com/JetBrains/kotlin/blob/9ee0d6b60ac4f0ea0ccc5dd01146bab92fabcdf2/core/descriptors/src/org/jetbrains/kotlin/types/TypeUtils.java#L455-L458
-          typeToResolve.original
-        }
-        else -> {
-          typeToResolve
-        }
-      }
-
-      return resolveGenericTypeReference(declaringClass, parameterKotlinType.toString())
-    }
-
-    /**
-     * Safely resolves a PSI [KtTypeReference], when that type reference may be a generic
-     * expressed by a type variable name. This is done by inspecting the class hierarchy to
-     * find where the generic type is declared, then resolving *that* reference.
-     *
-     * For instance, given:
-     *
-     * ```
-     * interface Factory<T> {
-     *   fun create(): T
-     * }
-     *
-     * interface ServiceFactory : Factory<Service>
-     * ```
-     *
-     * The KtTypeReference `T` will fail to resolve, since it isn't a type. This function will
-     * instead look to the `ServiceFactory` interface, then look at the supertype declaration
-     * in order to determine the type.
-     *
-     * @receiver The class which actually references the type. In the above example, this would
-     * be `ServiceFactory`.
-     * @param typeReference The generic reference to be resolved. In the above example, this
-     * is `T`.
-     */
-    public fun resolveTypeReference(typeReference: KtTypeReference): KtTypeReference? {
```

Comment:
That these functions are gone from this class makes so much more sense.

Reply:
> That these functions are gone from this class makes so much more sense.

This is super nice to see.